shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -44,10 +45,12 @@
   @NonNull
   private final DartMessenger messenger;
   private boolean isApplicationRunning = false;
+  private String isolateServiceId;
 
   public DartExecutor(@NonNull FlutterJNI flutterJNI) {
     this.flutterJNI = flutterJNI;
     this.messenger = new DartMessenger(flutterJNI);
+    messenger.setMessageHandler("flutter/isolate", isolateChannelMessageHandler);
```

Comment:
This is racy to changes in the engine that affect isolate initialization. This executor is initialized on the platform thread while the engine is racing to creating the isolate on the UI thread. It the engine wins, the message get dropped on the floor.

Reply:
Not sure I understand the concern here.  In the embedder `DartExecutor` will be created and the message handler will be installed at `FlutterNativeView`/`FlutterEngine` construction time.  This will happen before the app can call `runBundle` and start the isolate.

(Note that the message containing the isolate ID is being sent to the platform host side, not to Dart code)